### ISSUE TITLE
Desktop 's background is break after awake from suspend in AMD proprietary driver (fglrx)

### ISSUE THREAD

**Issue by dangchienhsgs** [linuxmint/Cinnamon#4747]
I tested on both LMDE Betsy and Linux Mint Cinnamon 17.2. Everytime I suspend and wake it, sometimes my background collapsed like this:
https://www.dropbox.com/s/uzobyrqxk228maa/Screenshot%20from%202015-10-17%2016%3A18%3A46.png?dl=0
Everything work well except I must change my wallpaper again to re-draw by desktop. This issue occurs at the first time I suspend my computer, the later times it does not occur.

**Comment by leigh123linux** [comment]
The screenshot shows a graphics driver issue not cinnamon so please close this bug and re-file it against fglrx instead!

**Comment by dangchienhsgs** [comment]
But in Ubuntu and Ubuntu Mate I don't get it so I think there is a relation between Linux Mint and it

**Comment by lestcape** [comment]
Unity and Mate don't use the same Gl functions of Cinnamon. So, if is working in them this not mean anything about where is the fault. What could say about where is the fault, is that is related with your specific hardware and this mean in most of cases isn't a cinnamon problem.

**Comment by dangchienhsgs** [comment]
Ok thank you very much :) I will close my issue

**Comment by lehjr** [comment]
Pretty sure the same issue was in Gnome and was claimed to be an FBO related issue. https://bugzilla.gnome.org/show_bug.cgi?id=739178

"This is a known interaction problem between the way GNOME stores its backgrounds and the fact that FBO contents are not preserved across suspend / resume cycles. In general, applications should avoid relying on FBO contents being preserved across mode switches or VT switches."

**Comment by dangchienhsgs** [comment]
Thanks for your link, I found that issue currently is not happening on Gnome 3. Should cinnamon have a fix like that ?

**Comment by lestcape** [comment]
@dangchienhsgs open the issue again. Is certainly a problem of nvidia architecture, but there are a way to workaround the problem.

**Comment by JosephMcc** [comment]
Not sure if the gnome problem has anything to do with Cinnamon or not. What I do know is that the two handle backgrounds very differently.

**Comment by lestcape** [comment]
@JosephMcc true, i don't say will be the same patch for cinnamon than for gnome, and i don't say will be the same problem, but the symptoms are really similar, so is possible that can be fixed in a similar way. In any case can be try, the gnome devs tried to solve the problem when they know was not his fault. The same attempt could be done in cinnamon i think.

**Comment by lestcape** [comment]
Please note that the gnome problem was for nvidia, and this is reported for AMD, but also note that the gnome patch also fixed an intel graphics. Apparently, what is needed is a background repaint after a awake for a suspended. Whatever was really the problem and whatever was the way or code used in it could be checked if a repaint fixed the problem.

**Comment by lehjr** [comment]
@letscape It's also mentioned in comment 15 that this issue also affects AMD devices. Kinda surprised to see Intel graphics in there though, maybe I'm still just not used to taking Intel graphics seriously.

**Comment by lestcape** [comment]
@lehjr you have right... See an intel was interesting to me i suppose. I'm not an expert, but there are some intel onboard graphics with acceleration.

**Comment by lehjr** [comment]
Btw, this was also reported here (maybe other locations as well): 
https://bugs.launchpad.net/linuxmint/+bug/1419729
https://bugs.launchpad.net/linuxmint/+bug/1433082

**Comment by nicdumz** [comment]
Just found what seems to be a duplicate issue^.

I'm affected too.
A google search yields a few other reports, https://bbs.archlinux.org/viewtopic.php?id=191219

The current title is a bit hard to parse, I suggest renaming this issue to `background corrupted on resume from suspend with AMD proprietary driver (fglrx)`

**Comment by ZanMarolt** [comment]
Hello, I am experiencing the same issue on my laptop AMD drivers. Is there a solution out there yet?

OS: LinuxMint 17.3 (rosa)
Gnome 3.8.4
Kernel 3.19.0-32-generic

**Comment by ZanMarolt** [comment]
I have found a workaround. After the suspend has jagged your background you can either click ALT + F2 and enter R. You can open up your terminal window and type `$ cinnamon --replace` or you can run this command each time on the startup and give it a slight delay to make sure your instance of cinnamon is already up and running already. 

Works for me :+1:
